packages/react/src/app-layout/AppLayoutSidebarNav.tsx

```diff
@@ -31,86 +34,146 @@ export type AppLayoutSidebarNavProps = {
+		<Stack as="nav" aria-label="main" paddingBottom={1.5}>
+			<Stack as="ul">
+				{items.map((group, index) => {
+					const isFirstItem = index === 0;
+					const groupItems = (Array.isArray(group) ? group : group.items).map(
+						addIsActive(activePath)
+					);
 
 					const disableGroupPadding = Array.isArray(group)
 						? false
 						: Boolean(group.options?.disableGroupPadding);
 
-					const prevGroup = items[idx - 1];
+					const prevGroup = items[index - 1];
 					const prevGroupDisableGroupPadding = prevGroup
 						? Array.isArray(prevGroup)
 							? false
 							: Boolean(prevGroup.options?.disableGroupPadding)
 						: false;
 
 					return (
-						<Fragment key={idx}>
+						<Fragment key={index}>
 							{!isFirstItem ? (
 								<AppLayoutSidebarNavDivider
 									// As the divider is placed at the start of each group, padding top is determined by the previous group
 									disablePaddingTop={prevGroupDisableGroupPadding}
 									disablePaddingBottom={disableGroupPadding}
 								/>
 							) : null}
-							{groupItems.map((item, idx) => (
-								<AppLayoutSidebarNavListItem
-									activePath={activePath}
-									background={background}
-									item={item}
-									key={idx}
-								/>
-							))}
+							{groupItems.map((item, index) => {
+								const isOpen =
+									subLevelVisible === 'always' ||
+									hasSubLevelActiveItem(item.items, activePath);
+
+								return (
+									<AppLayoutSidebarNavListItem
+										activePath={activePath}
+										background={background}
+										isOpen={isOpen}
+										item={item}
+										key={index}
+									/>
+								);
+							})}
 						</Fragment>
 					);
 				})}
-			</Flex>
-		</Flex>
+			</Stack>
+		</Stack>
 	);
-}
+};
 
 type AppLayoutSidebarNavListItemProps = {
 	activePath?: string;
 	background?: 'body' | 'bodyAlt';
+	isOpen: boolean;
 	item: NavItem;
 };
 
 function AppLayoutSidebarNavListItem({
 	activePath,
 	background,
+	isOpen,
 	item,
 }: AppLayoutSidebarNavListItemProps) {
 	const Link = useLinkComponent();
-	const { endElement, icon: Icon, label, ...restItemProps } = item;
+	const {
+		endElement,
+		icon: Icon,
+		isActive,
+		items,
+		label,
+		level,
+		...restItemProps
+	} = item;
 	const { closeMobileMenu } = useAppLayoutContext();
 
 	// Link list item
 	if ('href' in item) {
-		const active = item.href === activePath;
+		const isCurrentPage = item.href === activePath;
 		return (
 			<AppLayoutSidebarNavItemInner
 				background={background}
 				hasEndElement={Boolean(endElement)}
-				isActive={activePath === item.href}
+				isCurrentPage={isCurrentPage}
+				isActive={isCurrentPage || isOpen}
+				level={item.level}
 				onClick={closeMobileMenu} // Let the click event bubble up and close the menu on press of interactive item
 			>
-				<Link aria-current={active ? 'page' : undefined} {...restItemProps}>
-					{Icon ? <Icon color="inherit" /> : null}
+				<Link
```

Comment:
I'm not sure I know where the final design is, but it feels weird that we're not using the same chevron idea to indicate parents. How will users know when it's "whenActive"?

Reply:
Yeah, I believe that's where the badge goes:

![image](https://github.com/agriculturegovau/agds-next/assets/43688949/6517b09d-f393-44eb-9f2c-71c811ca1f06)

Though, I'll double check with design.